lib/calculator/parser.rb

```diff
@@ -63,7 +63,61 @@ def shift_tokens
 
     # Entry point for the parser, parses an <em>expression</em>
     def parse_expression
+
```

Comment:
Not sure if this was intentional or not, but you made all of the parse_X functions defined within parse_expression